state/upgrades.go

```diff
@@ -881,18 +877,30 @@ func SplitLogCollections(st *State) error {
 
 	// drop the old collection
 	if err := oldLogs.DropCollection(); err != nil {
-		// If the error is &mgo.QueryError{Code:26, Message:"ns not found", Assertion:false}
-		// that's fine.
-		if merr, ok := err.(*mgo.QueryError); ok {
-			if merr.Code == 26 {
-				return nil
-			}
+		// If the namespace is already missing, that's fine.
+		if isMgoNamespaceNotFound(err) {
+			return nil
 		}
 		return errors.Annotate(err, "failed to drop old logs collection")
 	}
 	return nil
 }
 
+func isMgoNamespaceNotFound(err error) bool {
+	// Check for &mgo.QueryError{Code:26, Message:"ns not found"}
+	if qerr, ok := err.(*mgo.QueryError); ok {
+		if qerr.Code == 26 {
```

Comment:
Off-hand, shouldn't we have a a name for the error code instead of just a number mgo.NamespaceNotFoundCode or somesuch?